r-storage-account.tf

```
@@ -103,5 +105,5 @@ resource "azurerm_role_assignment" "storage_account_blob_owner_current_user" {
   description          = "Temporary role assignment. Delete this assignment if unsure why it is still existing."
   principal_id         = local.init_access_azure_principal_id
   role_definition_name = "Storage Blob Data Owner"
-  scope                = azurerm_storage_account.this.id
+  scope                = local.resource_group_id
```

The resource group ID is already included in the storage account ID.  
You can extract it directly without introducing an extra local variable:

```suggestion
  scope                = split("/providers/", azurerm_storage_account.this.id)[0]
```